scylla-cql/src/frame/frame_errors.rs

```diff
@@ -61,6 +57,24 @@ pub enum ParseError {
     CqlTypeError(#[from] CqlTypeError),
 }
 
+/// An error type returned when deserialization of RESULT response fails.
+#[non_exhaustive]
+#[derive(Error, Debug)]
+pub enum CqlResultParseError {
+    #[error("Malformed RESULT response id: {0}")]
+    ResultIdParseError(LowLevelDeserializationError),
+    #[error("Unknown RESULT response id: {0}")]
+    UnknownResultId(i32),
+    #[error("'Set_keyspace' response deserialization failed: {0}")]
+    SetKeyspaceParseError(#[from] SetKeyspaceParseError),
+    #[error("'Schema_change' response deserialization failed: {0}")]
+    SchemaChangeEventParseError(#[from] SchemaChangeEventParseError),
```

Comment:
@wprzytula @muzarski Do you think that this variant should be present here?
CQL events are always sent on a separate channel (-1) which handles only that - so there should be no way for a user to get this error

Reply:
I think that you meant `CqlResponseParseError::CqlEventParseError` variant. I think that you are right. If so, then I'll remove this variant from public API.

Reply:
I see no easy way to decouple event handling from ordinary response handling. Thus, we need a common error type for both.

Reply:
It's true. Now the question is whether we want to introduce new public error type excluding this specific variant? I'm not sure if there is much sense to do so.

Reply:
I'd leave it as is for now.

Reply:
I see your point (why would handling events have its own error variants in QueryError), but the size of such refactor makes it out of this PR's scope IMO.

Reply:
It's not only about Events. `CqlResponseParseError` has the following variants:
```
    CqlErrorParseError
    CqlAuthChallengeParseError
    CqlAuthSuccessParseError
    CqlAuthenticateParseError
    CqlSupportedParseError
    CqlEventParseError
    CqlResultParseError
```
If I'm not mistaken only first and last are possible for the user to encounter in QueryError (only possible in NewSessionError). I understand that it may increase the amount of changes, but I think we should avoid creating such types anyway. It makes it much harder for users who want to handle errors in a different way than just printing them.

Reply:
OK, this is quite convincing. I suggest that @muzarski analyse all possible error routes in the driver and try to redesign the error hierarchy so that user-facing error types don't contain variants that the user is never supposed to see. If needed, we can introduce two layers of error handling:
- a lower-lever one, resembling one introduced in this PR,
- a higher-level one, restricting variants to those that a user really might see. It should be organised with the user's perspective in mind rather than reflect the driver's inner architecture.